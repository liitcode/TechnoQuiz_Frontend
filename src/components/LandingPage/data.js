export const textsection = {
  lightBg: false,
  lightText: true,
  lightTextDesc: true,
  topLine: 'Take a Quiz On',
  headline: 'Test your technical knowledge',
  description: [
    'JAVASCRIPT',
    'HTML',
    'DOCKER',
    'PHP',
    'KUBERNETES',
    'MYSQL',
    'WORDPRESS',
    'LINUX',
    'DEVOPS',
    'BASH',
    'LARAVEL',
  ],
  imgStart: '',
  img: 'images/5.svg',
  alt: 'TechnoQuiz',
};
export const sectionOne = {
  lightBg: true,
  lightText: false,
  lightTextDesc: false,
  topLine: 'TechnoQuiz',
  headline: 'Interactive way to learn',
  description: 'Enhance your skills and expand your knowledge. Get the best of us!',
  buttonLabel: 'Sign Up',
  imgStart: 'start',
  img: 'images/1.svg',
  alt: 'TechnoQuiz',
};

export const sectionTwo = {
  lightBg: false,
  lightText: true,
  lightTextDesc: true,
  topLine: 'Product',
  headline: 'What is it all about?',
  description:
     'Choose your favourite topics with a variety of questions set for beginners and seniors alike!',
  buttonLabel: 'Try Now',
  imgStart: '',
  img: 'images/2.svg',
  alt: 'TechnoQuiz',
};
export const sectionThree = {
  lightBg: true,
  lightText: false,
  lightTextDesc: false,
  topLine: 'Premium',
  headline: 'Buy our Premium Membership to get an extra edge',
  description: 'Get acccess to all hard questions and explainations',
  buttonLabel: 'View Plans',
  imgStart: 'start',
  img: 'images/3.svg',
  alt: 'TechnoQuiz',
};

export const sectionFour = {
  lightBg: false,
  lightText: true,
  lightTextDesc: true,
  topLine: 'TechnoQuiz',
  headline: 'Learning Made Easy',
  description: 'Hone your technical skills',
  buttonLabel: 'Explore Now',
  imgStart: '',
  img: 'images/4.svg',
  alt: 'TechnoQuiz',
};

export const videoSectionOne = {
  headline: 'Test Your Skills',
  description: 'choose your favourite programming language',
  buttonLabel: 'Explore Now',
  vid: 'videos/2.mp4',
};
