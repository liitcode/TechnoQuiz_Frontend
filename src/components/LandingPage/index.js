/* eslint-disable react/jsx-props-no-spreading */
import React,{ useEffect } from 'react';
import Aos from 'aos';
import Section from '../UI/TwoUpSection';
import TextSection from '../UI/TextSection';
import { sectionOne,sectionTwo,sectionThree,textsection } from './data';

function Home(){
    useEffect(()=>{
        Aos.init({duration: 1000});
    },[]);

    return(
        <>
        {/* <VideoSection {...videoSectionOne} /> */}
        <TextSection {...textsection}/>
        <Section {...sectionOne}/>
        <Section {...sectionTwo}/>
        <Section {...sectionThree}/>
        </>
    )
}

export default Home