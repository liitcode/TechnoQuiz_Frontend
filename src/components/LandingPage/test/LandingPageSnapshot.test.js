import React from "react";
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import storeConfig from '``../../../Redux/store/configureStore';
import LandingPage from "../index";

it('renders the component', () => {
    const tree = renderer.create(<Provider store={storeConfig}>
        <MemoryRouter><LandingPage /></MemoryRouter></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})