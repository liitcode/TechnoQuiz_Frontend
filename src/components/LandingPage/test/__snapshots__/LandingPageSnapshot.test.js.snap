// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
Array [
  <div
    className="home__section darkBg"
  >
    <div
      className="container"
    >
      <div
        className="row home__section-row"
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <div
          className="col"
        >
          <div
            className="home__section-text-wrapper"
          >
            <div
              className="heading"
            >
              Test your technical knowledge
            </div>
            <div
              className="top-line"
            >
              Take a Quiz On
            </div>
            <h1
              className="heading"
            >
              <div
                className="Typewriter"
                data-testid="typewriter-wrapper"
              />
            </h1>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="home__section-img-wrapper"
          >
            <img
              alt="TechnoQuiz"
              className="home__section-img"
              src="images/5.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="home__section"
  >
    <div
      className="container"
    >
      <div
        className="row home__section-row"
        style={
          Object {
            "display": "flex",
            "flexDirection": "row-reverse",
          }
        }
      >
        <div
          className="col"
        >
          <div
            className="home__section-text-wrapper"
          >
            <div
              className="top-line"
            >
              TechnoQuiz
            </div>
            <h1
              className="heading_dark"
            >
              Interactive way to learn
            </h1>
            <p
              className="home__section-subtitle dark"
            >
              Enhance your skills and expand your knowledge. Get the best of us!
            </p>
            <a
              href="/signup"
              onClick={[Function]}
            >
              <button
                className="btn btn--primary btn--wide yellow"
              >
                Sign Up
              </button>
            </a>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="home__section-img-wrapper"
          >
            <img
              alt="TechnoQuiz"
              className="home__section-img"
              data-aos="fade-right"
              src="images/1.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="home__section darkBg"
  >
    <div
      className="container"
    >
      <div
        className="row home__section-row"
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <div
          className="col"
        >
          <div
            className="home__section-text-wrapper"
          >
            <div
              className="top-line"
            >
              Product
            </div>
            <h1
              className="heading"
            >
              What is it all about?
            </h1>
            <p
              className="home__section-subtitle"
            >
              Choose your favourite topics with a variety of questions set for beginners and seniors alike!
            </p>
            <a
              href="/category"
              onClick={[Function]}
            >
              <button
                className="btn btn--primary btn--wide yellow"
              >
                Try Now
              </button>
            </a>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="home__section-img-wrapper"
          >
            <img
              alt="TechnoQuiz"
              className="home__section-img"
              data-aos="fade-left"
              src="images/2.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="home__section"
  >
    <div
      className="container"
    >
      <div
        className="row home__section-row"
        style={
          Object {
            "display": "flex",
            "flexDirection": "row-reverse",
          }
        }
      >
        <div
          className="col"
        >
          <div
            className="home__section-text-wrapper"
          >
            <div
              className="top-line"
            >
              Premium
            </div>
            <h1
              className="heading_dark"
            >
              Buy our Premium Membership to get an extra edge
            </h1>
            <p
              className="home__section-subtitle dark"
            >
              Get acccess to all hard questions and explainations
            </p>
            <a
              href="/premium"
              onClick={[Function]}
            >
              <button
                className="btn btn--primary btn--wide yellow"
              >
                View Plans
              </button>
            </a>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="home__section-img-wrapper"
          >
            <img
              alt="TechnoQuiz"
              className="home__section-img"
              data-aos="fade-right"
              src="images/3.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
